lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -271,10 +271,17 @@ LogicalResult ChannelBufferLowering::matchAndRewrite(
     numStages = stages.getValue().getLimitedValue();
   }
   Value input = buffer.input();
+  StringAttr bufferName = buffer.options().name();
   for (uint64_t i = 0; i < numStages; ++i) {
     // Create the stages, connecting them up as we build.
     auto stage = rewriter.create<PipelineStage>(loc, type, buffer.clk(),
                                                 buffer.rstn(), input);
+    if (bufferName) {
+      std::string stageName;
+      llvm::raw_string_ostream(stageName)
```

Comment:
I think for constructing a string like this, you can use `+` with a `StringRef` instead of creating a `raw_string_ostream` directly. Are you able to say `bufferName.getValue() + "_stage" + std::to_string(i)`? I think that is mildly preferable if it works.

Reply:
No, that didn't work but I did find a better way with the SmallString constructor.

Reply:
If you want to get fancy, you can also investigate [`llvm::Twine`](https://llvm.org/docs/ProgrammersManual.html#the-twine-class)

Reply:
https://llvm.org/docs/ProgrammersManual.html#the-twine-class